File: app/src/main/java/com/eftimoff/bakingapp/app/images/VideoThumbnailLoader.java
package com.eftimoff.bakingapp.app.images;

import android.support.annotation.Nullable;

import com.bumptech.glide.load.Options;
import com.bumptech.glide.load.model.ModelLoader;
import com.bumptech.glide.signature.ObjectKey;

import java.io.InputStream;

public class VideoThumbnailLoader implements ModelLoader<VideoThumbnailUrl, InputStream> {

    @Nullable
    @Override
    public LoadData<InputStream> buildLoadData(VideoThumbnailUrl videoThumbnailUrl, int width, int height, Options options) {
        return new LoadData<>(new ObjectKey(videoThumbnailUrl), new VideoThumbnailFetcher(videoThumbnailUrl));
    }

    @Override
    public boolean handles(VideoThumbnailUrl videoThumbnailUrl) {
        return true;
    }
}

File: app/src/main/java/com/eftimoff/bakingapp/app/images/VideoThumbnailLoader.kt
package com.eftimoff.bakingapp.app.images

import com.bumptech.glide.load.Options
import com.bumptech.glide.load.model.ModelLoader
import com.bumptech.glide.signature.ObjectKey
import java.io.InputStream


class VideoThumbnailLoader : ModelLoader<VideoThumbnailUrl, InputStream> {
    override fun handles(url: VideoThumbnailUrl): Boolean = true

    override fun buildLoadData(videoThumbnailUrl: VideoThumbnailUrl, width: Int, height: Int, options: Options?): ModelLoader.LoadData<InputStream> {
        return ModelLoader.LoadData(ObjectKey(videoThumbnailUrl), VideoThumbnailFetcher(videoThumbnailUrl))
    }
}